src/engine/BMDB.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * BMDB: database interface to standardize types 
+ *
+ * @author: Chaos
+ */
+
+/**
+ * This class contains all the logic to do with selecting from and updating
+ * the buttonmen database tables.
+ *
+ * It is not intended to be instantiated --- it's a set of public functions
+ * that callers will invoke using their own database connection.
```

Comment:
This comment isn't true any more --- i changed my mind while writing the code, and now the class is instantiated with an open database connector.  IMO even better would be if we could eventually move the connector into this class, and just invoke it with `$isTest` or whatever.  That's because i don't like the thing where if a database connection temporarily fails (e.g. because the database is being restarted as part of a routine nightly restart), we throw exceptions to users.  The only reason we can't catch failures where `$conn` is undefined and retry connecting, is that the connection logic is so scattered.  So we might as well plan to have `BMDB` eventually own all of the connection management, because then it will be able to do friendly things like that.